### DISCUSSION TITLE
How to set custom route name and path in Nuxt 3?

### DISCUSSION THREAD

**Discussion by Trixpua** [nuxt/nuxt#17349] [Questions] [answered]
In Nuxt 2 I usually keep the files inside page with a english name and change the route name and path using the `'router-extras-module'` and it works fine, I tried use the `'pages:extend'` hook in Nuxt 3 to achieve it, but the path in the browser don't change only act as an alias, in the below example I can push the route by the route name (site.about) or directly by the path (/sobre), but the url in browser keep the filename (/about). There is a way to change the path permanently?

Other question, there is way to change the name and path in each page file instead of the change in the nuxt.config.ts?

nuxt.config.ts

```
    hooks: {
        'pages:extend'(pages) {
            pages.push({
                name: 'site.about',
                path: '/sobre',
                file: resolve(__dirname, './pages/about/index.vue'),
            });
        },
    },
```

*.vue

`<NuxtLink :to="{ name: 'site.about' }">link</NuxtLink>`

**Comment by ROBJkE** [comment, marked as answer]
First of all thanks for that example, really helped me to set custom paths and override the original.
So i can't confirm that 'alias' behavior in rc4.

``` js
'pages:extend': (pages) => {
  pages.push({
    name: 'portfolio',
    path: '/my-awesome-projects',
    file: '@/pages/portfolio.vue',
  });
}
```

This replace the `/portfolio` path with `/my-awesome-projects`.
If i try to call `/portfolio` now i get a 404 as it should be.
And `<NuxtLink :to="{ name: 'portfolio }">link</NuxtLink>` resolves to `/my-awesome-projects`.
All testet with Nuxt3 rc4.

Edit: The original Nuxt generated pages still remains in the `pages` array.  So maybe its better to replace paths for convenience.

`nuxt.config.ts`
``` js 
// import some custom routes
import { routes } from "./data/seoRoutes.js";

export default defineNuxtConfig({
...
  hooks: {
    'pages:extend': (pages) => {

      pages.map((page) => {
 
        routes.forEach(seoRoute => {
          if (seoRoute.name === page.name) {
            Object.assign(page, {path: seoRoute.path});
          }
        })
      })
    }
  },
...
})

```

`seoRoutes.js`
``` js 
export const routes = [{
    name: 'portfolio',
    path: '/my-awesome-projects',
}, {
    name: 'about',
    path: '/about-my-work',
}]
```

**Reply by Trixpua** [reply]
I don't know if I made some mistake before or it changed in the new release, I upgraded to rc4 and now it is working properly. Thank you
